stickermanager.py

```diff
@@ -0,0 +1,276 @@
+        if not event.is_reply:
+            return
+        elif current_vote:
+            await event.reply('There\'s already an ongoing sticker poll.')
+            return
+
+        async with current_vote_lock:
+            await start_poll_int(event)
+
+    async def start_poll_int(event: Union[events.NewMessage.Event, Message]) -> None:
+        global current_vote
+
+        if current_vote:
+            await event.reply('There\'s already an ongoing sticker poll.')
+            return
+
+        emoji = event.pattern_match.group(1) or '\u2728'
+        try:
+            _, sender_name = WEIGHTS[event.sender_id]
+        except KeyError:
+            return
+        orig_evt: Message = await event.get_reply_message()
+        # TODO add support for animated stickers
+        if not orig_evt.photo and (not orig_evt.sticker or
+                                   orig_evt.sticker.mime_type == 'application/x-tgsticker'):
+            return
+
+        filename = os.path.join(os.path.dirname(__file__), f'stickermanager.{int(time())}.dat')
+        await orig_evt.download_media(filename)
+
+        asyncio.ensure_future(event.delete(), loop=bot.loop)
+        current_vote = {
+            'chat': event.chat_id,
+            'sender_id': event.sender_id,
+            'sender_name': sender_name,
+            'score': 0,
+            'emoji': emoji,
+            'votes': {},
+            'filepath': filename,
+            'pack_link': f'https://t.me/addstickers/{STICKER_PACK_SHORT_NAME}',
+            'animated': orig_evt.sticker and orig_evt.sticker.mime_type == 'application/x-tgsticker'
+        }
+        reply_evt: Message = await orig_evt.reply(
+            POLL_TEMPLATE.format_map(get_template_data()),
+            buttons=[Button.inline(UP, b'+'), Button.inline(DOWN, b'-')],
+            parse_mode='html')
+        current_vote['poll'] = reply_evt.id
+        asyncio.ensure_future(reply_evt.pin(notify=True), loop=bot.loop)
+
+    @bot.on(events.CallbackQuery(chats=ALLOWED_CHATS, data=lambda data: data in (b'+', b'-')))
+    async def vote_poll(event: events.CallbackQuery.Event) -> None:
+        if not current_vote or current_vote['poll'] != event.message_id:
+            await event.answer('That poll is closed.')
+            return
+
+        async with current_vote_lock:
+            await vote_poll_int(event)
+
+    async def vote_poll_int(event: events.CallbackQuery.Event) -> None:
+        global current_vote
+
+        if not current_vote or current_vote['poll'] != event.message_id:
+            await event.answer('That poll is closed.')
+            return
+
+        weight, displayname = WEIGHTS.get(event.sender_id, (DEFAULT_WEIGHT, None))
+        if weight == 0:
+            await event.answer('You don\'t have the permission to vote.')
+            return
+        if event.data == b'-':
+            weight = -weight
+
+        displayname = displayname or escape_html((await event.get_sender()).first_name)
```

Comment:
Note that `utils.get_display_name` exists, but perhaps using only the first name is more appropriated.

Reply:
Yeah, people in the whitelist also only have first names

Reply:
I don't 😏

Reply:
I mean the whitelist only has first names, you're "twit"